compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/checkers/declaration/FirGetterSetterChecker.kt

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2010-2021 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.fir.analysis.checkers.declaration
+
+import org.jetbrains.kotlin.fir.analysis.checkers.context.CheckerContext
+import org.jetbrains.kotlin.fir.analysis.diagnostics.DiagnosticReporter
+import org.jetbrains.kotlin.fir.analysis.diagnostics.FirErrors.WRONG_SETTER_PARAMETER_TYPE
+import org.jetbrains.kotlin.fir.analysis.diagnostics.reportOn
+import org.jetbrains.kotlin.fir.declarations.FirProperty
+import org.jetbrains.kotlin.fir.types.coneType
+
+object FirGetterSetterChecker : FirPropertyChecker() {
```

Comment:
Maybe `FirPropertyAccessorChecker` sounds better.

But, I'm not sure whether we need yet another checker. How about adding a routine to `FirDeclarationCheckerUtils#checkPropertyAccessors`: https://github.com/JetBrains/kotlin/blob/master/compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/checkers/declaration/FirDeclarationCheckerUtils.kt#L172 which will be invoked per property/accessors?

Reply:
I'll add another diagnostics in this checker, e.g. GETTER_VISIBILITY_DIFFERS_FROM_PROPERTY_VISIBILITY, SETTER_VISIBILITY_INCONSISTENT_WITH_PROPERTY_VISIBILITY, etc. Therefore, I think, i need another checker.

Reply:
Renamed to FirPropertyAccessorChecker manually

Reply:
Instead, I'm moving `FirDeclarationCheckerUtils#checkPropertyAccessors` to `FirPropertyAccessorChecker`: https://github.com/JetBrains/kotlin/pull/4279